### PR TITLE
Vocab support

### PR DESCRIPTION
**Pull request by vivekNEC**

### CODE REVIEW

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java [hunk 1]
```diff
@@ -595,6 +604,7 @@ private void validateAttribute(int payloadType, String expandedProperty, String
 			return;
 		}
 		if (isScalar) {
+			System.out.println("testing "+Constants.allowedScalars.get(payloadType).contains(expandedProperty));
```
Comment (ScorpioBroker) [inline Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java]: remove

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]
lgtm just remove the sysout

**Review by ScorpioBroker — APPROVED** [review]

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java [hunk 2]
```diff
@@ -42,6 +42,11 @@ class NGSIObject {
 	private boolean hasAtContext = false;
 	private boolean atContextRequired = false;
 
+
```
Comment (amitrnec) [inline Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:45]: Remove spec

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java [hunk 3]
```diff
@@ -1032,9 +1046,34 @@ private int commonWherePart(String[] attribPath, String[] subAttribPath, String
...
+				sql.append(NGSIConstants.NGSI_LD_VocabularyProperty);
+				sql.append("' THEN EXISTS (SELECT TRUE FROM JSONB_ARRAY_ELEMENTS(");
+				sql.append(currentSqlAttrib);
+				sql.append(" ->'");
+				sql.append(NGSIConstants.NGSI_LD_HAS_VOCAB);
+				sql.append("') AS mostInnerValue WHERE (mostInnerValue->'");
+				sql.append(NGSIConstants.JSON_LD_ID);
+				sql.append("')");
+     			dollarCount = applyOperator(sql, dollarCount, tuple,true);
+//				for (int i=0;i<tuple.size();i++){
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java:1065]: Remove comment code.

**Pull request merged**